crates/nu_plugin_query/src/query_xml.rs

```diff
@@ -88,7 +88,8 @@ pub fn execute_xpath_query(
 
             match r {
                 sxd_xpath::Value::Nodeset(ns) => {
-                    for n in ns.into_iter() {
+                    let nodes = ns.document_order();
+                    for n in nodes {
```

Comment:
I don't think there is a reason to create an extra `nodes` variable.   Can you just change it to this?
```rust
for n in ns.document_order()
```